### PR TITLE
[FIX] hr_holidays: Fix leave out of office message

### PR DESCRIPTION
**Pull request by LucasLefevre**
1/ Editable after approve
========

Purpose
-----------
Currently, the ooom field is editable in the leave view but an ir.rule
prevents the employee to save as soon as the leave is approved.

Specification
-----------------
An employee should be able to edit the out of office message
even if the leave is already approved.

2/ Default value
============
If a user has an ooom in his Preferences, it should be
used as the default value for a leave ooom.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_holidays/models/hr_leave.py [hunk 1]
```diff
@@ -572,6 +577,11 @@ def _read_from_database(self, field_names, inherited_field_names=[]):
 
     @api.multi
     def write(self, values):
+        # Allow an employee to always write his own out of office message
+        if len(self) == 1 and values.keys() == {'out_of_office_message'} and self.employee_id.user_id == self.env.user:
+            values = {'out_of_office_message': values.get('out_of_office_message')}
```
Comment (RomainLibert) [inline addons/hr_holidays/models/hr_leave.py]: I believe that this line is not really useful as you can only get here if you write only `out_of_office_message`, hence this line is just copying `values`.

Comment (LucasLefevre) [reply addons/hr_holidays/models/hr_leave.py]: True. Fixed

**Comment by RomainLibert** [comment]
@robodoo r+

**Comment by RomainLibert** [comment]
@robodoo rebase-ff

**Comment by tbe-odoo** [comment]
Hello guys,

When merging fixes that requires an upgrade (`-u`) you should ping one of the SaaS team members so we can apply the fix correctly.
(When we update our prod database, we only checkout the sources)

**Comment by LucasLefevre** [comment]
@tbe-odoo Ok, sorry
Thanks for the reminder

**Pull request closed without merging**